Assets/Scripts/Combatscripts/Abilities/GenericAbilityStrategy.cs

```diff
@@ -186,9 +189,33 @@ private void ApplyDamage(PlayerController pcTarget, ObstacleController obstTarge
         }
     }
 
+    private void ApplyHailEffect(MechStats.AbilityMechSlot slot, PlayerController character, GameObject tileTarget)
+    {
+        if (CanHailBeUsed(slot, character, tileTarget))
+        {
+            /*
+             * TODO:
+             * 1. Move the enemy-now-friendly unit to the player's side
+             *      (sidenote - i just need to change a boolean and remove a script for them to be playable.
+             *      a system checks to see which units have these characteristics to determine if it is an enemy)
+             * 2. Get list of all the enemy units in the scene
+             * 3. Increase hail resistance value for all of them
+             *      (sidenote - this could be determined by some random value maybe?)
+             * 4. Increase the amount of recruitment points based on how high the unit's resistance value was. The
+             *    higher the resistance, the more exp you gain from the hailing. 
+             */
+        }
+    }
 
-
-
+    // Decides if an enemy can be hailed. Determines threshold using random int between 0 and 100 (inclusive). 
+    private bool CanHailBeUsed(MechStats.AbilityMechSlot slot, PlayerController character, GameObject tileTarget)
+    {
+        var possibleTarget = tileTarget.GetComponent<PlayerController>();
+        bool canTargetBeHailed = possibleTarget.pilotInfo.GetHailResistance() < (Random.Range(0, 100));
```

Comment:
Technically this can fail if you have a hail resistance of 0 and the random number is 0